pennylane/math/quantum.py

```diff
@@ -247,31 +247,40 @@ def reduce_dm(density_matrix, indices, check_state=False, c_dtype="complex128"):
     return _permute_dense_matrix(density_matrix, sorted(indices), indices, batch_dim)
 
 
-def _batched_partial_trace(density_matrix, indices):
+def partial_trace(matrix, indices, c_dtype="complex128"):
     """Compute the reduced density matrix by tracing out the provided indices.
 
     Args:
-        density_matrix (tensor_like): 3D density matrix tensor. This tensor should be of size
-            ``(batch_dim, 2**N, 2**N)``, for some integer number of wires``N``.
+        matrix (tensor_like): 2D or 3D density matrix tensor. For a 2D tensor, the size is assumed to be
+            ``(2**n, 2**n)``, for some integer number of wires``n``.
+            ``(2**n, 2**n)``
+
         indices (list(int)): List of indices to be traced.
 
     Returns:
-        tensor_like: (reduced) Density matrix of size ``(batch_dim, 2**len(wires), 2**len(wires))``
+        tensor_like: (reduced) Density matrix of size ``(2**len(wires), 2**len(wires))``
 
     **Example**
+    >>> x = np.array([[1, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]])
+    >>> partial_trace(x, indices=[0])
+    array([[1, 0], [0, 0]])
+
+    >>> x = np.array([[1, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]])
+    >>> partial_trace(x, indices=[0])
+    array([[1, 0], [0, 0]])
 
     >>> x = np.array([[[1, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]],
-    ...               [[0, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]]])
-    >>> _batched_partial_trace(x, indices=[0])
+                     [[0, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]]])
+    >>> partial_trace(x, indices=[0])
     array([[[1, 0],
             [0, 0]],
 
            [[0, 0],
             [0, 1]]])

```

Comment:
This renders a mess (not your fault as the old doc string was never rendered, but you should check the rendered version now that it becomes a public/user facing function :))

![image](https://github.com/PennyLaneAI/pennylane/assets/43949391/b6ebfe1a-f7d9-4bb4-9ed3-f0a530659788)

The output is actually correct, its just that the full line break gets interpreted as the code block being finished. Instead you could do the suggestion below (not sure this will work, you will have to check the rendered documentation)
```
.. code-block:: python3
    >>> x = np.array([
            [[1, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]],
            [[0, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]]
        ])
    >>> partial_trace(x, indices=[0])
    array([[[1, 0],
            [0, 0]],

           [[0, 0],
            [0, 1]]])
```
Here it is important that everything inside the "code-bock" environment has the correct indentation and that you leave at least one linebreak after

Reply:
pro-tip: if you want to play with it and render docs locally you can do `make docs` in the repository (be sure to install all requirements in `docs/requirements.txt`

Reply:
@Qottmann So I was unable to get the code-blocks to work for me but i did manage to clean up the documentation and the code blocks with proper line breaks and indentations, as shown [here](https://xanaduai-pennylane--5152.com.readthedocs.build/en/5152/code/api/pennylane.math.partial_trace.html).

One thing that I'm a bit confused by is that in the successfully built documentation, partial_trace was listed at the bottom even though in pennylane/doc/conf.py on line 316 `autodoc_member_order = "bysource"`. Wondering if there's somewhere else where that ordering scheme is getting overriden?

![Screenshot 2024-02-17 at 10 19 07 PM](https://github.com/PennyLaneAI/pennylane/assets/39506616/70e481e8-4aa6-4238-8ba3-fc92a1c36df4)